sycl/include/sycl/detail/common.hpp

```diff
@@ -491,7 +491,7 @@ template <typename DataT, template <typename, typename> typename FlattenF,
 struct ArrayCreator<DataT, FlattenF, ArgT, ArgTN...> {
   static constexpr auto Create(const ArgT &Arg, const ArgTN &...Args) {
     auto ImmArray = FlattenF<DataT, ArgT>()(Arg);
-    if constexpr (sizeof...(Args))
+    if constexpr (sizeof...(Args) > 0)
```

Comment:
Note for reviewers: strictly speaking, the existing code is valid and warning is a bug in MSVC: https://developercommunity.visualstudio.com/t/false-positive-warning-when-using-if-constexpr/842830, but we still need to support older MSVC (2019) so that's why the change is being made

Reply:
It would be better if have this in the code comments instead of PR comments.

Reply:
Added a comment in https://github.com/intel/llvm/pull/8611.